code/frontend/src/components/Answer/Answer.tsx

```diff
@@ -24,23 +31,51 @@ export const Answer = ({
 }: Props) => {
     const [isRefAccordionOpen, { toggle: toggleIsRefAccordionOpen }] = useBoolean(false);
     const filePathTruncationLimit = 50;
-
+    const answerContainerRef =  useRef<HTMLDivElement>(null);// read the text from the container
     const messageBoxId = "message-" + index;
-
+    const [isSpeaking, setIsSpeaking] = useState(false); // for speaker on
+    const [showSpeaker, setShowSpeaker] = useState(true); //for show and hide the speaker icon
+    const [isPaused, setIsPaused] = useState(false); //for pause
     const parsedAnswer = useMemo(() => parseAnswer(answer), [answer]);
     const [chevronIsExpanded, setChevronIsExpanded] = useState(isRefAccordionOpen);
     const refContainer = useRef<HTMLDivElement>(null);
+    const [audioContext, setAudioContext] = useState<AudioContext | null>(null); //Manully  manage the audio context eg pausing resuming
+    // const [synthesizer, setSynthesizer] = useState<sdk.SpeechSynthesizer | null>(null);
     const handleChevronClick = () => {
         setChevronIsExpanded(!chevronIsExpanded);
         toggleIsRefAccordionOpen();
       };
 
+    const fetchSpeechConfig = async (): Promise<{ key: string; region: string }> => {
```

Comment:
These changes look unrelated to the PR. Are they requried for something else?

Reply:
reverted the made changes